drivers/adc/adc_ltc2451.c

```diff
@@ -0,0 +1,104 @@
+/* LLTC LTC2451 ADC
+ *
+ * Copyright (c) 2023 Brill Power Ltd.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/adc.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(ltc2451, CONFIG_ADC_LOG_LEVEL);
+
+#define DT_DRV_COMPAT lltc_ltc2451
+
+struct ltc2451_config {
+	struct i2c_dt_spec i2c;
+	uint8_t conversion_speed;
+};
+
+static int ltc2451_channel_setup(const struct device *dev,
+				const struct adc_channel_cfg *channel_cfg)
+{
+	ARG_UNUSED(dev);
+
+	if (channel_cfg->channel_id != 0) {
+		LOG_ERR("Invalid channel id '%d'", channel_cfg->channel_id);
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+static int ltc2451_set_conversion_speed(const struct device *dev, uint8_t conversion_speed)
+{
+	const struct ltc2451_config *config = dev->config;
+	uint8_t wr_buf[1];
+	int err;
+
+	if (conversion_speed == 60) {
+		wr_buf[0] = 0;
+	} else if (conversion_speed == 30) {
+		wr_buf[0] = 1;
+	} else {
+		LOG_ERR("Invalid conversion speed selected");
+		return -EINVAL;
+	}
+
+	err = i2c_write_dt(&config->i2c, wr_buf, sizeof(wr_buf));
+
+	if (err != 0) {
+		LOG_ERR("LTC write failed (err %d)", err);
+	}
+
+	return err;
+}
+
+static int ltc2451_read_latest_conversion(const struct device *dev,
+					  const struct adc_sequence *sequence)
+{
+	const struct ltc2451_config *config = dev->config;
+	uint8_t rd_buf[2];
+	uint16_t *value_buf;
+	int err = i2c_read_dt(&config->i2c, rd_buf, sizeof(rd_buf));
+
+	if (err == 0) {
+		value_buf = (uint16_t *)sequence->buffer;
+		value_buf[0] = (rd_buf[0] << 8) | rd_buf[1];
```

Comment:
What about endianness here? Shouldn't be `sys_get_be16` used?

Reply:
Yeah my bad, has been updated!